R/ancova.R

```diff
@@ -797,6 +799,851 @@ Ancova <- function(jaspResults, dataset = NULL, options) {
+  if (comparison == "unconstrained")
+    nms <- c(modelNames, gettext("Unconstrained"))
+  else if (comparison == "complement")
+    nms <- c(modelNames, gettext("Complement"))
+  else
+    nms <- modelNames
+
+  compareDf <- compareGoric[["result"]]
+  compareDf[["model"]] <- nms
+
+  if (type == "goric")
+    weights <- compareDf[["goric.weights"]]
+  else
+    weights <- compareDf[["gorica.weights"]]
+
+  ratio <- weights/weights[nms == reference]
+  compareDf[["ratio"]] <- ratio
+
+  if (length(nms) == 1)
+    comparisonTable$addRows(as.list(compareDf))
+  else
+    comparisonTable$addColumns(as.list(compareDf))
+
+  return()
+}
+
+.anovaOrdinalRestrictionsCreateCoefficientsTable <- function(compareGoric, modelNames, ordinalRestrictionsContainer, options) {
+  if (!is.null(ordinalRestrictionsContainer[["coefficientsTable"]])) return()
+
+  coefTable <- createJaspTable(gettext("Model Coefficients Table"))
+  coefTable$dependOn(c("restrictedModelComparison", "restrictedModelComparisonCoefficients", "highlightEstimates"))
+  overtitle <- gettext("Estimates")
+  coefTable$addColumnInfo(name = "coef", title = gettext("Coefficient"), type = "string")
+
+  for (name in modelNames) {
+    coefTable$addColumnInfo(name = name, title = name, type = "number", overtitle = overtitle)
+  }
+
+  compareAgainst <- options[["restrictedModelComparison"]]
+
+  if (compareAgainst == "complement")
+    coefTable$addColumnInfo(name = "complement", title = gettext("Complement"), type = "number", overtitle = overtitle)
+
+  coefTable$addColumnInfo(name = "unconstrained", title = gettext("Unconstrained"), type = "number", overtitle = overtitle)
+
+  ordinalRestrictionsContainer[["coefficientsTable"]] <- coefTable
+
+  if (ordinalRestrictionsContainer$getError()) return()
+
+  .anovaOrdinalRestrictionsCoefficientsTableFill(coefTable, compareGoric, modelNames, compareAgainst, options[["includeIntercept"]], options[["highlightEstimates"]])
+
+  return()
+}
+
+.anovaOrdinalRestrictionsCoefficientsTableFill <- function(coefTable, compareGoric, modelNames, compareAgainst, includeIntercept, highlightEstimates) {
+
+  xLevels    <- compareGoric[["model.org"]][["xlevels"]]
+  coefs      <- coef(compareGoric)
+  coefNames  <- names(coefs)
+  terms      <- strsplit(coefNames, ":")
+
+  if (compareAgainst != "unconstrained")
+    coefs    <- rbind(coefs, coef(compareGoric[["model.org"]]))
+
+  for (i in 1:length(xLevels)) {
+    idxFac  <- which(stringr::str_detect(coefNames, names(xLevels)[i]))
+    idxLvl  <- if (includeIntercept || i > 1) 1:(length(xLevels[[i]])-1) else 1:length(xLevels[[i]])
+    newLvls <- numeric(length(idxFac))
+    newLvls[1:length(idxFac)] <- idxLvl
+
+    for (j in 1:length(idxFac)) {
+      idxInt <- which(stringr::str_detect(terms[[idxFac[j]]], names(xLevels)[i]))
+
+      terms[[idxFac[j]]][idxInt] <- paste0(names(xLevels)[i], " (", newLvls[j], ")")
+    }
+  }
+
+  coefNamesPretty  <- sapply(terms, paste, collapse = " \u273B ")
```

Comment:
```suggestion
  coefNamesPretty  <- sapply(terms, paste, collapse = jaspBase::interactionSymbol)
```
Note that this uses thin spaces (`\u2009`) instead of spaces.

Reply:
thnks